src/jit/lowerxarch.cpp

```diff
@@ -2603,19 +2617,73 @@ bool Lowering::IsContainableHWIntrinsicOp(GenTreeHWIntrinsic* containingNode, Ge
             assert(supportsAlignedSIMDLoads == false);
             assert(supportsUnalignedSIMDLoads == false);
 
-            supportsSIMDScalarLoads = true;
-            supportsGeneralLoads    = supportsSIMDScalarLoads;
+            switch (containingIntrinsicId)
+            {
+                case NI_Base_Vector128_CreateScalarUnsafe:
+                case NI_Base_Vector256_CreateScalarUnsafe:
+                {
+                    assert(supportsSIMDScalarLoads == false);
+
+                    const unsigned expectedSize = genTypeSize(genActualType(containingNode->gtSIMDBaseType));
+                    const unsigned operandSize  = genTypeSize(node->TypeGet());
+
+                    supportsGeneralLoads = (operandSize == expectedSize);
```

Comment:
Why is here `==` and others `>=` ?

Reply:
Because this is getting the size of the actual type for `expectedSize` (so `TYP_BYTE` and `TYP_SHORT` will return 4).

Reply:
Oh, just saw your comment https://github.com/dotnet/coreclr/pull/21641#issuecomment-452056255. Thanks.

Maybe someones still can be `>=`. For example, `ConvertScalarToVector128Int32((int)longVal)` could be a direct truncation?

Reply:
Agree, thanks.